### PR TITLE
Add specs for microsoft.monitor RP

### PR DESCRIPTION
**Pull request by divya-jay**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [x] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. - _Currently deployed in Public regions_
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month. - _August_
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+            "read"
+          ]
+        }
+      }
+    },
+    "IngestionSettings": {
+      "description": "Settings for data ingestion",
+      "type": "object",
+      "properties": {
+        "dataCollectionRule": {
```
Comment (raosuhas) [inline specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json:367]: >dataCollectionRule [](http://example.com/codeflow?start=9&length=18)

consider endoing this name with "...resourceId"

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+          "description": "The Azure resource Id of the default data collection rule for this monitoring account.",
+          "type": "string",
+          "readOnly": true,
+          "x-ms-mutability": [
+            "read"
+          ]
+        },
+        "dataCollectionEndpoint": {
```
Comment (raosuhas) [inline specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json:375]: >dataCollectionEndpoint [](http://example.com/codeflow?start=9&length=22)

same here for resourceid

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+            "read"
+          ]
+        },
+        "provisioningState": {
+          "description": "The provisioning state of the monitoring account. Set to Succeeded if everything is healthy.",
+          "type": "string",
+          "readOnly": true,
+          "x-ms-mutability": [
+            "read"
+          ]
```
Comment (raosuhas) [inline specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json:415]: Enum

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json [hunk 4]
```diff
@@ -0,0 +1,556 @@
...
+        },
+        "tags": {
+          "description": "Resource tags",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "id": {
```
Comment (raosuhas) [inline specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json:462]: cosnider using trackedresource from commontypes for this

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/examples/MonitoringAccountsCreate.json [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        },
+        "properties": {
+          "accountId": "2df515bf-c3ce-4920-84d4-1d9d16542d9f",
+          "metrics": {
+            "prometheusQueryEndpoint": "https://myMonitoringAccount-v8hx.eastus.prometheus.monitor.azure.com",
+            "internalId": "mac_2df515bf-c3ce-4920-84d4-1d9d16542d9f"
+          },
+          "provisioningState": "Succeeded",
+          "defaultIngestionSettings": {
+            "dataCollectionRule": "/subscriptions/703362b3-f278-4e4b-9179-c76eaf41ffc2/resourceGroups/MA_myMonitoringAccount_eastus_managed/providers/Microsoft.Insights/dataCollectionRules/myMonitoringAccount",
```
Comment (raosuhas) [inline specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/examples/MonitoringAccountsCreate.json:30]: >subscriptions [](http://example.com/codeflow?start=36&length=13)

Doesnt the user need to specify these properties ? are they created implicitly ?

**Comment by raosuhas** [comment]
Please respond to the comments and remove the "ARMChangesRequested" label when this PR is ready for review again .

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/examples/MonitoringAccountsCreate.json (hunk 5 above)
Comment (divya-jay) [reply specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/examples/MonitoringAccountsCreate.json]: Yes, this is created by the RP automatically, in a managed Resource Group

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json (hunk 4 above)
Comment (divya-jay) [reply specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json]: Thanks for pointing me at that, makes this look much simpler

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json (hunk 3 above)
Comment (divya-jay) [reply specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json:425]: Updated, with the possible values

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json (hunk 2 above)
Comment (divya-jay) [reply specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json]: Renamed this property

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json (hunk 1 above)
Comment (divya-jay) [reply specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json]: Renamed this property

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/operations_API.json [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    }
+  },
+  "tags": [
+    {
+      "name": "Operations",
+      "description": "List available operations API"
+    }
+  ],
+  "paths": {
+    "/providers/Microsoft.Monitor/operations": {
```
Comment (zizw123) [inline specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/operations_API.json:49]: >operations [](http://example.com/codeflow?start=34&length=10)

How are these operations being created? Only list is supported?

Comment (divya-jay) [reply specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/operations_API.json:49]: This is per the RPC (https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#exposing-available-operations)

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json (hunk 3 above)
Comment (mentat9) [reply specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json:425]: Needs to include Canceled value. Ref: [https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html?q=provisioningstate#rpc005-provisioning-state-semantics-must-be-followed](https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html?q=provisioningstate#rpc005-provisioning-state-semantics-must-be-followed).

Comment (divya-jay) [reply specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json:426]: Thanks for pointing this out Chris, added Canceled as a possible state

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json [hunk 7]
```diff
@@ -0,0 +1,518 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Monitoring Accounts Control Plane API",
+    "description": "Provides operations for working with Azure Monitoring Accounts",
```
Comment (rkmanda) [inline specification/monitor/resource-manager/Microsoft.Monitor/preview/2021-06-03-preview/monitoringAccounts_API.json:5]: looks like this is a new service that is not being onboarded to the RP as a service platform. Can you arrange a meeting with Evan Hissey and Vikash Jain and discuss the reasons why you cannot be on the RpaaS platform? If you already have an exception to not be on the RpaaS platform please send an email to the armapireview alias attaching the exception email.

**Comment by divya-jay** [comment]
@ArcturusZhang can you help merge this PR into master?

**Comment by ArcturusZhang** [comment]
THis PR only adds new contents, the breaking changes must be false alarms

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**